src/parser/Expression.ts

```diff
@@ -336,8 +338,8 @@ export class FunctionParameterExpression extends Expression {
         super();
     }
 
-    public getType() {
-        return this.typeExpression?.getType() || DynamicType.instance;
+    public getType(flags: SymbolTypeFlags) {
+        return this.typeExpression?.getType(SymbolTypeFlags.typetime) ?? this.defaultValue?.getType(SymbolTypeFlags.runtime) ?? DynamicType.instance;
```

Comment:
Prefer type Expression, if it exists, otherwise initial value, otherwise it's Dynamic